### PR TITLE
feat(components): JOB-80773 Datatable scroll to top pagination loading

### PR DESCRIPTION
**Pull request by timothyong8**
## Motivations

- We would like the scroll on the `DataTable` to go back to the top when switching between pages as it feels strange to have to do that when presenting the user with a new page of data.

### Added

- `Pagination` component now takes in a `onPageChanged` function which will do the scroll

## Testing

- move the scroll to the bottom of a page and flip pages in both directions it should return to the top in each case

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/DataTable/DataTable.test.tsx
Comment (darryltec) [inline packages/components/src/DataTable/DataTable.test.tsx:1]: you should test your scroll bits and see if it's actually being called for precaution, y'know heh.

**Review by darryltec — CHANGES_REQUESTED** [review]
This is good! I think we can simplify the changes tho

File: packages/components/src/DataTable/DataTable.tsx [hunk 1]
```diff
@@ -87,7 +87,9 @@ export function DataTable<T extends object>({
   emptyState,
   loading = false,
 }: DataTableProps<T>) {
-  const [ref, { exactWidth }] = useResizeObserver();
+  const ref = useRef<HTMLDivElement>(null);
+
+  const [, { exactWidth }] = useResizeObserver({ ref });
```
[the remark this answers is not in the document]
Comment (timothyong8) [reply packages/components/src/DataTable/DataTable.tsx:92]: We tried to use the ref here but discovered [this](https://www.npmjs.com/package/use-resize-observer#getting-the-raw-element-from-the-default-refcallback) 

Note that "ref" in the above examples is a RefCallback, not a RefObject, meaning you won't be able to access "ref.current" if you need the element itself.

[the remark this answers is not in the document]
Comment (timothyong8) [reply packages/components/src/DataTable/DataTable.tsx:92]: I swear this was not working before but you are right.

File: packages/components/src/DataTable/Pagination.tsx [hunk 2]
```diff
@@ -7,10 +7,11 @@ import { Text } from "../Text";
...
 interface PaginationProps<T> {
-  table: Table<T>;
-  itemsPerPage?: number[];
-  totalItems: number;
-  loading: boolean;
+  readonly table: Table<T>;
+  readonly itemsPerPage?: number[];
+  readonly totalItems: number;
+  readonly loading: boolean;
+  readonly scrollRef: React.RefObject<HTMLElement>;
```
[the remark this answers is not in the document]
Comment (timothyong8) [reply packages/components/src/DataTable/Pagination.tsx:14]: Good idea!

**Review by darryltec — APPROVED** [review]
:shipit:

**Pull request merged**